docs/smart-contracts/entrypoints.md

```diff
@@ -12,77 +12,66 @@ The entrypoints in a Tezos smart contract must meet these specifications:
 - Each entrypoint must have a name.
 - Entrypoints may accept parameters, which can be of almost any data type that Tezos supports.
 
-Unlike functions and API endpoints, entrypoints do not return a value.
-To return a value from a smart contract, see [Views](./views).
+Unlike functions and API endpoints, entrypoints do not return a value directly to the caller.
+To return a value from a smart contract, you can use one of these methods:
 
-For examples of contracts, see [Examples of contracts](https://opentezos.com/smart-contracts/simplified-contracts/) on opentezos.com.
+- Use [Views](./views)
+- Include a callback parameter that sends information to another smart contract, as in the `getAllowance`, `getBalance`, and `getTotalSupply` entrypoints of [FA1.2](../architecture/tokens/FA1.2) contracts
+
+For an example of a simple contract, see the tutorial [Create a smart contract](../tutorials/smart-contract).
+
+For examples of more complex contracts, see [Examples of contracts](https://opentezos.com/smart-contracts/simplified-contracts/) on opentezos.com.
 
 ## Entrypoint logic
 
-An entrypoint may run all kinds of logic based on:
+An entrypoint may run logic based on:
 
 - Its storage
```

Comment:
contract storage. An entrypoint does not have it sown storage, it is global to the contract

Reply:
Thanks -- updated.